api/src/transactions.rs

```diff
@@ -508,99 +562,192 @@ impl TransactionsApi {
-        ledger_info: LedgerInfo,
         txn: SignedTransaction,
-    ) -> SubmitTransactionResult<PendingTransaction> {
+    ) -> poem::Result<PendingTransaction, AptosError> {
         let (mempool_status, vm_status_opt) = self
             .context
             .submit_transaction(txn.clone())
             .await
             .context("Mempool failed to initially evaluate submitted transaction")
             .map_err(|err| {
-                SubmitTransactionError::internal_with_code(
-                    err,
-                    AptosErrorCode::InternalError,
-                    &ledger_info,
-                )
+                aptos_api_types::AptosError::new_with_error_code(err, AptosErrorCode::InternalError)
             })?;
         match mempool_status.code {
-            MempoolStatusCode::Accepted => match accept_type {
-                AcceptType::Json => {
-                    let resolver = self.context.move_resolver_poem(&ledger_info)?;
-                    let pending_txn = resolver
-                            .as_converter(self.context.db.clone())
-                            .try_into_pending_transaction_poem(txn)
-                            .context("Failed to build PendingTransaction from mempool response, even though it said the request was accepted")
-                            .map_err(|err| SubmitTransactionError::internal_with_code(
-                                err,
-                                AptosErrorCode::InternalError,
-                                &ledger_info,
-                            ))?;
-                    SubmitTransactionResponse::try_from_json((
-                        pending_txn,
-                        &ledger_info,
-                        SubmitTransactionResponseStatus::Accepted,
-                    ))
-                }
-                AcceptType::Bcs => SubmitTransactionResponse::try_from_bcs((
-                    (),
-                    &ledger_info,
-                    SubmitTransactionResponseStatus::Accepted,
-                )),
-            },
+            MempoolStatusCode::Accepted => {
+                let resolver = self
+                    .context
+                    .move_resolver()
+                    .context("Failed to read latest state checkpoint from DB")
+                    .map_err(|e| {
+                        AptosError::new_with_error_code(e, AptosErrorCode::InternalError)
+                    })?;
+
+                let pending_txn = resolver
+                        .as_converter(self.context.db.clone())
+                        .try_into_pending_transaction_poem(txn)
+                        .context("Failed to build PendingTransaction from mempool response, even though it said the request was accepted")
+                        .map_err(|err| AptosError::new_with_error_code(
+                            err,
+                            AptosErrorCode::InternalError,
+                        ))?;
+                Ok(pending_txn)
```

Comment:
Wait a minute, this gets rid of the performance wins by not converting the pending transaction at the end.  We don't need to convert the pending transaction we can just return the hash.

Reply:
ah, now I get what you were doing, sorry. moving this to json processing part.